presto-main/src/main/java/com/facebook/presto/server/ResourceUtil.java

```diff
@@ -203,6 +214,33 @@ public static void assertRequest(boolean expression, String format, Object... ar
         }
     }
 
+    private static void parsePreparedStatementsHeader(String header, Map<String, String> preparedStatements)
+    {
+        List<String> nameValue = Splitter.on('=').limit(2).trimResults().splitToList(header);
+        assertRequest(nameValue.size() == 2, "Invalid %s header", PRESTO_PREPARED_STATEMENT);
+
+        String statementName;
+        String sqlString;
+        try {
+            statementName = URLDecoder.decode(nameValue.get(0), "UTF-8");
+            sqlString = URLDecoder.decode(nameValue.get(1), "UTF-8");
+        }
+        catch (UnsupportedEncodingException e) {
```

Comment:
`UnsupportedEncodingException` is only thrown if the encoding is not supported. This can be re-thrown as `AssertionError` since `UTF-8` is guaranteed to be supported.

We do need to catch `IllegalArgumentException` since that is thrown if the input data is invalid.

```java
catch (UnsupportedEncodingException e) {
    throw new AssertionError(e);
}
catch (IllegalArgumentException e) {
    throw badRequest(format("Invalid %s header: %s", PRESTO_PREPARED_STATEMENT, e.getMessage()));
}